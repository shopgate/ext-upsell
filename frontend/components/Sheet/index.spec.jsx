import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

const mockedStore = configureStore();
// eslint-disable-next-line react/prop-types, require-jsdoc
const MockedSheetComponent = props => (<div id="sheet">{props.children}</div>);
jest.mock('@shopgate/pwa-ui-shared/Sheet', () => MockedSheetComponent);

let mockedProductRelationsFiltered = [];
jest.mock('../../selectors', () => ({
  getProductRelationsFiltered: () => () => mockedProductRelationsFiltered,
}));
// eslint-disable-next-line require-jsdoc
const MockedGridComponent = () => <div>Hello world</div>;
jest.mock('../../components/Grid', () => MockedGridComponent);

jest.mock('@shopgate/pwa-extension-kit/env/helpers', () => ({
  isIOSTheme: () => false,
}));

describe('Sheet', () => {
  // eslint-disable-next-line global-require
  const Sheet = require('./index').default;
  const defaultProps = {
    headline: 'Mocked headline',
    isOpen: false,
    onClose: () => {},
    productId: 'mockedId',
    showName: true,
    showPrice: true,
    type: 'mockedType',
    titleRows: 3,
    maxItemsPerLine: 3,
  };

  it('should render with 3 items per line as default', () => {
    mockedProductRelationsFiltered = [1, 2, 3];
    const component = mount((
      <Provider store={mockedStore({})}>
        <Sheet {...defaultProps} />
      </Provider>
    ));
    expect(component.find(MockedGridComponent).props().itemsPerLine).toBe(3);
    expect(component).toMatchSnapshot();
  });

  it('should render with 2 items per line', () => {
    mockedProductRelationsFiltered = [1, 2];
    const component = mount((
      <Provider store={mockedStore({})}>
        <Sheet {...defaultProps} />
      </Provider>
    ));
    expect(component.find(MockedGridComponent).props().itemsPerLine).toBe(2);
    expect(component).toMatchSnapshot();
  });

  it('should render with 1 item per line', () => {
    mockedProductRelationsFiltered = [1];
    const component = mount((
      <Provider store={mockedStore({})}>
        <Sheet {...defaultProps} />
      </Provider>
    ));
    expect(component.find(MockedGridComponent).props().itemsPerLine).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('should keep the sheet closed when there are no items to show', () => {
    mockedProductRelationsFiltered = [];
    const component = mount((
      <Provider store={mockedStore({})}>
        <Sheet {...defaultProps} isOpen />
      </Provider>
    ));
    // Check if mocking actually works.
    expect(component.find(Sheet).props().isOpen).toBe(true);
    // Check if SheetComponent is still closed.
    expect(component.find(MockedSheetComponent).props().isOpen).toBe(false);
    expect(component).toMatchSnapshot();
  });
});
