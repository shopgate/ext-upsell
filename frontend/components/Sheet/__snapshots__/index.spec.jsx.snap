// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sheet should keep the sheet closed when there are no items to show 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Sheet)
    headline="Mocked headline"
    isOpen={true}
    onClose={[Function]}
    productId="mockedId"
    showName={true}
    showPrice={true}
    type="mockedType"
  >
    <Sheet
      dispatch={[Function]}
      headline="Mocked headline"
      isOpen={true}
      onClose={[Function]}
      productId="mockedId"
      productsCount={0}
      showName={true}
      showPrice={true}
      type="mockedType"
    >
      <MockedSheetComponent
        className="css-dctx2g"
        contentClassName="css-19olcfv"
        isOpen={false}
        onClose={[Function]}
        title="Mocked headline"
      >
        <div
          id="sheet"
        >
          <MockedGridComponent
            itemsPerLine={3}
            productId="mockedId"
            showName={true}
            showPrice={true}
            type="mockedType"
          >
            <div>
              Hello world
            </div>
          </MockedGridComponent>
        </div>
      </MockedSheetComponent>
    </Sheet>
  </Connect(Sheet)>
</Provider>
`;

exports[`Sheet should render with 1 item per line 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Sheet)
    headline="Mocked headline"
    isOpen={false}
    onClose={[Function]}
    productId="mockedId"
    showName={true}
    showPrice={true}
    type="mockedType"
  >
    <Sheet
      dispatch={[Function]}
      headline="Mocked headline"
      isOpen={false}
      onClose={[Function]}
      productId="mockedId"
      productsCount={1}
      showName={true}
      showPrice={true}
      type="mockedType"
    >
      <MockedSheetComponent
        className="css-domgpv"
        contentClassName="css-1vybkpy"
        isOpen={false}
        onClose={[Function]}
        title="Mocked headline"
      >
        <div
          id="sheet"
        >
          <MockedGridComponent
            itemsPerLine={1}
            productId="mockedId"
            showName={true}
            showPrice={true}
            type="mockedType"
          >
            <div>
              Hello world
            </div>
          </MockedGridComponent>
        </div>
      </MockedSheetComponent>
    </Sheet>
  </Connect(Sheet)>
</Provider>
`;

exports[`Sheet should render with 2 items per line 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Sheet)
    headline="Mocked headline"
    isOpen={false}
    onClose={[Function]}
    productId="mockedId"
    showName={true}
    showPrice={true}
    type="mockedType"
  >
    <Sheet
      dispatch={[Function]}
      headline="Mocked headline"
      isOpen={false}
      onClose={[Function]}
      productId="mockedId"
      productsCount={2}
      showName={true}
      showPrice={true}
      type="mockedType"
    >
      <MockedSheetComponent
        className="css-dctx2g"
        contentClassName="css-19olcfv"
        isOpen={false}
        onClose={[Function]}
        title="Mocked headline"
      >
        <div
          id="sheet"
        >
          <MockedGridComponent
            itemsPerLine={2}
            productId="mockedId"
            showName={true}
            showPrice={true}
            type="mockedType"
          >
            <div>
              Hello world
            </div>
          </MockedGridComponent>
        </div>
      </MockedSheetComponent>
    </Sheet>
  </Connect(Sheet)>
</Provider>
`;

exports[`Sheet should render with 3 items per line as default 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Sheet)
    headline="Mocked headline"
    isOpen={false}
    onClose={[Function]}
    productId="mockedId"
    showName={true}
    showPrice={true}
    type="mockedType"
  >
    <Sheet
      dispatch={[Function]}
      headline="Mocked headline"
      isOpen={false}
      onClose={[Function]}
      productId="mockedId"
      productsCount={3}
      showName={true}
      showPrice={true}
      type="mockedType"
    >
      <MockedSheetComponent
        className="css-dctx2g"
        contentClassName="css-19olcfv"
        isOpen={false}
        onClose={[Function]}
        title="Mocked headline"
      >
        <div
          id="sheet"
        >
          <MockedGridComponent
            itemsPerLine={3}
            productId="mockedId"
            showName={true}
            showPrice={true}
            type="mockedType"
          >
            <div>
              Hello world
            </div>
          </MockedGridComponent>
        </div>
      </MockedSheetComponent>
    </Sheet>
  </Connect(Sheet)>
</Provider>
`;
